goblet/infrastructures/infrastructure.py

```diff
@@ -27,6 +28,12 @@ def register(self, name, kwargs):
         raise NotImplementedError("register")
 
     def deploy(self):
+        if not self.supports_local:
```

Comment:
this is what is causing infra to be skipped. @nthings 

should only skip if not supports local and is deploying locally.